server/app/models/ApiKey.java

```diff
@@ -32,67 +36,107 @@ public ApiKey setGrants(ApiKeyGrants grants) {
     return this;
   }
 
-  @DbJsonB private ApiKeyGrants grants;
-
+  /** Timestamp of when the the ApiKey was created. */
   public Instant getCreateTime() {
     return createTime;
   }
 
+  /** Timestamp of when the the ApiKey was last modified. */
   public Instant getUpdateTime() {
     return updateTime;
   }
 
+  /** Timestamp of when the ApiKey is no longer valid. */
   public Instant getExpiration() {
     return expiration;
   }
 
+  /**
+   * Timestamp of when the ApiKey is no longer valid. Expiration should be immutable after creation.
+   */
   public ApiKey setExpiration(Instant expiration) {
     this.expiration = expiration;
     return this;
   }
 
+  /** The {@code authorityId} of the account that created the ApiKey. */
   public String getCreatedBy() {
     return createdBy;
   }
 
+  /** The {@code authorityId} of the account that created the ApiKey. */
   public ApiKey setCreatedBy(String createdBy) {
     this.createdBy = createdBy;
     return this;
   }
 
+  /** Human readable name of the ApiKey. */
   public String getName() {
     return name;
   }
 
+  /** Human readable name of the ApiKey. OK to modify after creation. */
   public ApiKey setName(String name) {
     this.name = name;
     return this;
   }
 
+  /**
+   * Unique identifier for the ApiKey. Paired with the password, comprises the credentials for using
+   * the ApiKey. For more information see https://en.wikipedia.org/wiki/Basic_access_authentication
+   */
   public String getKeyId() {
     return keyId;
   }
 
+  /** Unique identifier for the ApiKey. */
   public ApiKey setKeyId(String keyId) {
     this.keyId = keyId;
     return this;
   }
 
+  /**
+   * The salted key secret (a.k.a. password) of the ApiKey. Created by signing the password with the
+   * API secret using the SHA-HMAC-256 algorithm.
+   */
   public String getSaltedKeySecret() {
     return saltedKeySecret;
   }
 
+  /** The salted key secret (a.k.a. password) of the ApiKey. */
   public ApiKey setSaltedKeySecret(String saltedKeySecret) {
     this.saltedKeySecret = saltedKeySecret;
     return this;
   }
 
+  /**
+   * An allowlist of IPv4 addresses that are permitted to authenticate with this ApiKey. Specified
+   * using CIDR notation: https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing
+   */
   public String getSubnet() {
     return subnet;
   }
 
+  /**
+   * An allowlist of IPv4 addresses that are permitted to authenticate with this ApiKey. Specified
+   * using CIDR notation: https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing
+   */
   public ApiKey setSubnet(String subnet) {
     this.subnet = subnet;
     return this;
   }
+
+  /**
+   * The client IPv4 address of the last request to successfully auth with the ApiKey. Empty if the
+   * ApiKey has never been used.
+   */
+  public Optional<String> getLastCallIpAddress() {
```

Comment:
Is this meant to be material to the API or more like light logging for dev testing/curiosity?

Reply:
The latter, also useful for light security / auditing purposes.